torch/csrc/jit/interpreter.cpp

```diff
@@ -193,6 +362,55 @@ struct CodeImpl {
   std::vector<bool> bool_data;
 };
 
+
+struct Registers {
+  Registers(size_t size)
+  : registers(size) {}
+  Registers(const Registers & rhs)
+  : registers(rhs.registers) {
+    for(auto & r : registers) {
+      if(isValid(r))
+        r->retain();
+    }
+  }
+  ~Registers() {
+    for(auto & r : registers) {
+      if(isValid(r))
+        r->release();
+    }
+  }
+  at::RefCounted*& operator[](size_t i) {
```

Comment:
Returning a reference in here seems dangerous - if you have a managed object in `i`th position, and you just overwrite its pointer with another one, then the previous one won't be freed.

At least describing the intended semantics would be helpful.

Reply:
Ill try to refactor so that we have owning/non-owning implementations.